collie/movielens/get_data.py

```diff
@@ -258,3 +300,54 @@ def get_movielens_metadata(df_item: pd.DataFrame = None) -> pd.DataFrame:
     metadata_df = metadata_df[cols]
 
     return metadata_df
+
+
+def get_user_metadata(df_user: pd.DataFrame = None) -> pd.DataFrame:
+    """
+    Return MovieLens 100K user metadata as a DataFrame.
+
+    DataFrame returned has the following column order:
+
+    .. code-block:: python
+
+        [
+            'age', 'female', 'occupation_administrator', 'occupation_artist'
+            'occupation_doctor', 'occupation_educator', 'occupation_engineer'
+            'occupation_entertainment', 'occupation_executive'
+            'occupation_healthcare', 'occupation_homemaker'
+            'occupation_lawyer', 'occupation_librarian', 'occupation_marketing'
+            'occupation_none', 'occupation_other', 'occupation_programmer'
+            'occupation_retired', 'occupation_salesman', 'occupation_scientist'
+            'occupation_student', 'occupation_technician', 'occupation_writer',
+        ]
+
+    See the MovieLens 100K README for additional information on the dataset:
+    https://files.grouplens.org/datasets/movielens/ml-100k-README.txt
+
+    Parameters
+    ----------
+    df_user: pd.DataFrame
+        DataFrame of MovieLens 100K ``u.user`` containing columns of user
+        metadata. If ``None``, will automatically read the output of ``read_movielens_df_user()``
+
+    Returns
+    -------
+    metadata_df: pd.DataFrame
+
+    """
+    if df_user is None:
+        df_user = read_movielens_df_user()
+
+    # format user occupation
+    df_user_occupation = df_user.iloc[:, [3]].copy()
+    df_occupation = pd.get_dummies(df_user_occupation.occupation, prefix='occupation')
+    df_occupation = df_occupation.sort_index(axis=1)
```

Comment:
What does this line do?

Reply:
it sorts the occupations in alphabetical order, it's not really necessary, more of a nice to have

Reply:
Ah, that makes sense. I like that!